test/semantics/separate_module_procs_2.f90

```diff
@@ -0,0 +1,41 @@
+! Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+module m1
+  interface ma
+  end interface
+end module
+
+submodule (m1) ma_submodule
```

Comment:
`m1` should declare a separate module procedure or it can't be used as the ancestor module of a submodule. We don't detect this error yet, but should some day.

Reply:
I was not aware of this requirement. At least the documents that i read about sub modules, did not mention it. We can raise a new issue to handle this scenario.
I wrote this test case to show the error condition of not defining separate module procedure.
Tim, Could you please share any document which explains about this requirement?

Reply:
It follows directly from the standard.  If a module declares no separate module procedure, its submodules can have no effect on the program.  (See C.9.4(2): "A separate module procedure that is declared in a module to have public accessibility can be accessed by use association even if it is defined in a submodule. No other entity in a submodule can be accessed by use association.")

Put another way: if a submodule and its descendants define no separate module procedures apart from the SMPs that they themselves declare, they're all useless.  If a module declares no SMP, all of its submodules must be useless as a consequence.

Reply:
The main document you should be using is the Fortran 2018 standard. It says:
> C1412 (R1418) The ancestor-module-name shall be the name of a nonintrinsic module that declares a separate module procedure; ...

I'm not asking you to add that check, just to fix your test program so that the only errors in it are the ones you intend to be there.